querier.go

```diff
@@ -783,6 +783,9 @@ func newChunkSeriesIterator(cs []chunks.Meta, dranges Intervals, mint, maxt int6
 
 func (it *chunkSeriesIterator) Seek(t int64) (ok bool) {
 	if t > it.maxt {
+		it.i = len(it.chunks)
+	}
+	if it.cur.Err() != nil || it.i >= len(it.chunks) {
 		return false
 	}
```

Comment:
so here we are first advancing and than failing. Normally my logic would be that if an operation failed it shouldn't change the internal state.

Reply:
We're not failing, we're simply advancing to the requested timestamp and returning `false`, meaning there are no more values.

If we would not advance, then a `Seek` before `maxt` following a `Seek` past `maxt` would incorrectly position the iterator on a valid value.

Reply:
Agree with @free  and adding exactly same thing to https://github.com/prometheus/tsdb/pull/665